src/cspviolationdetector/DefaultCSPViolationDetector.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import CSPViolationDetector from './CSPViolationDetector';
+
+export default class DefaultCSPViolationDetector implements CSPViolationDetector {
+  private static count: number = 0;
+
+  /* istanbul ignore next */
+  private static start(): void {
+    if (!('document' in global) || !document.addEventListener) {
+      return;
+    }
+    if (DefaultCSPViolationDetector.count === 0) {
+      document.addEventListener('securitypolicyviolation', DefaultCSPViolationDetector.listener);
+    }
+    DefaultCSPViolationDetector.count += 1;
+  }
```

Comment:
Why jump through all of these hoops rather than always installing the listener unless the builder flips a flag off?

Reply:
OK, I will re-write this part for disabling the detector.